dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ExecutionModelNext.java

```diff
@@ -0,0 +1,134 @@
+package com.dat3m.dartagnan.verification.model;
+
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.filter.Filter;
+import com.dat3m.dartagnan.program.memory.MemoryObject;
+import com.dat3m.dartagnan.program.Thread;
+import com.dat3m.dartagnan.verification.model.event.*;
+import com.dat3m.dartagnan.verification.model.relation.RelationModel;
+import com.dat3m.dartagnan.wmm.Relation;
+
+import java.math.BigInteger;
+import java.util.*;
+
+
+// This is a new implementation of ExecutionModel which serves as the data structure
+// representing an execution in Dartagnan. It contains instances of EventModel for events
+// and RelationModel for relations. It is used only by ExecutionGraphVisualizer so far.
+public class ExecutionModelNext {
+    private final List<ThreadModel> threadList;
+    private final List<EventModel> eventList;
+    private final Map<Event, EventModel> eventMap;
+    private final Map<Relation, RelationModel> relationMap;
+    private final Map<MemoryObject, MemoryObjectModel> memoryLayoutMap;
+
+    private final Map<BigInteger, Set<LoadModel>> addressReadsMap;
+    private final Map<BigInteger, Set<StoreModel>> addressWritesMap;
+    private final Map<BigInteger, Set<MemoryEventModel>> addressAccessesMap;
+    private final Map<Thread, List<List<EventModel>>> atomicBlocksMap;
+
+    ExecutionModelNext() {
+        threadList = new ArrayList<>();
+        eventList = new ArrayList<>();
+        eventMap = new HashMap<>();
```

Comment:
Wouldn't it be cleaner to
- use a ByMap for the later (so we can easily implement `getEventModelById`)
- get rid of `eventList` and simply use the range/values of the map?

EDIT: I think you mentioned something to me why the list was indeed needed. But I don't really remember what for.

Reply:
I think a list makes sense to have the events ordered. That being said, one could also use a `LinkedHashMap` to retain order in a set.